importer/lambda/cross_region_importer.py

```diff
@@ -113,41 +118,42 @@ def _create_new_cross_stack_references(requested_exports, importer_context, tabl
     return response_data
 
 
-def _update_cross_stack_references(requested_exports, old_exports, importer_context, table_info):
-    requested_exports_labels = set(requested_exports.keys())
-    old_exports_labels = set(old_exports.keys())
-
-    export_labels_to_add = requested_exports_labels - old_exports_labels
-    export_labels_to_remove = old_exports_labels - requested_exports_labels
-
-    exports_to_add = {label_to_add: requested_exports[label_to_add] for label_to_add in export_labels_to_add}
-    exports_to_remove = {label_to_remove: old_exports[label_to_remove] for label_to_remove in export_labels_to_remove}
-
-    exports = _get_cloudformation_exports(table_info.target_region)
-
-    try:
-        response_data = {
-            label: exports[export_name]['Value'] for label, export_name in requested_exports.items()
-        }
-    except KeyError as e:
-        raise ExportNotFoundError(e.args[0])
-
-    _create_new_cross_stack_references(exports_to_add, importer_context, table_info)
-    _delete_cross_stack_references(exports_to_remove, importer_context, table_info)
-
-    return response_data
-
-
-def _delete_cross_stack_references(exports_to_remove, importer_context, table_info):
+def _delete_cross_stack_references(exports_to_remove, importer_context, table_info, physical_resource_id):
     dynamodb_resource = boto3.resource('dynamodb', region_name=table_info.target_region)
     cross_stack_ref_table = dynamodb_resource.Table(table_info.table_name)
 
     for label, export_name in exports_to_remove.items():
-        cross_stack_ref_id = f'{export_name}|{importer_context.stack_id}|{importer_context.logical_resource_id}|{label}'
+        cross_stack_ref_id = f'{physical_resource_id}|{export_name}'
         print(f'Removing cross-stack ref: {cross_stack_ref_id}')
-        cross_stack_ref_table.delete_item(
-            Key={'CrossStackRefId': cross_stack_ref_id},
-        )
+        try:
+            cross_stack_ref_table.delete_item(
+                Key={'CrossStackRefId': cross_stack_ref_id},
+                ConditionExpression=Attr('CrossStackRefId').eq(cross_stack_ref_id),
+            )
+        except ClientError:
```

Comment:
ClientError can raised for many reasons (bad request, throttling, etc). You should make sure the `ClientError` you're catching has the proper error code you're expecting. If not, you should re-raise the exception.